policies/rudderc/docs/src/usage.md

```diff
@@ -0,0 +1,105 @@
+# Usage
+
+## The `rudderc` CLI
+
+Rudder comes with a tool dedicated to the
+techniques development and usage.
+It is especially important as technique are not run as YAML, but compiled into
+an executable policy file depending on the target platform.
+There are currently two possible target, which are the platforms Rudder has agents for:
+
+* Linux/AIX
+* Windows
+
+These platforms use different agent technology but the YAML policies
+allows unifying them.
+To be able to check and compile technique, the `rudderc` program
+needs access to the methods library of the target systems.
+
+To get access to the base Linux methods set, you can use git
+and use the repository as library argument:
+
+```shell
+$ git clone https://github.com/Normation/ncf/
+# [...]
+$ rudderc subcommand -l /.../ncf/tree/30_generic_methods/ 
+```
+
+### Create a technique
+
+To setup the technique structure:
+
+```shell
+$ rudderc new my_technique
+       Wrote ./my_technique/technique.yml
+
+$ cd my_technique
+```
+
+This will create the base structure of your new technique:
+
+```text
+my_technique/
+  ├── technique.yml
+  └── resources/
+```
+
+The `technique.yml` is the technique content,
+and the `resources` directory can be used to include
+external files (configuration files, templates, etc.).
+All file produced by `rudderc` will be placed in the `target`
+directory.
+
+### Checking a technique
+
+You can check the current technique syntax with:
+
+```shell
+$ rudderc check --library /path/to/methods/lib
+        Read 179 methods (/path/to/methods/lib)
+   Compiling my_technique v0.1 [Linux]
+   Compiling my_technique v0.1 [Windows]
+     Checked technique.yml
+```
+
+This will check the technique schema and check the compilation
```

Comment:
Is it possible to add the schema in the doc with a link here?
Also, a small text to how to add the jsonchema to a text editor could be great.

Reply:
yes, but I plan to do this later as is needs some testing.

Reply:
but I'm adding the link

Reply:
Good :+1: